broker.go

```diff
@@ -361,11 +382,22 @@ func (b *Broker) muLeaderConnection(topic string, partition int32) (conn *connec
 			}
 			nodeID, ok = b.metadata.endpoints[tp]
 			if !ok {
-				b.conf.Logger.Info("cannot get leader connection: unknown topic or partition",
-					"topic", topic,
-					"partition", partition,
-					"endpoint", tp)
 				err = proto.ErrUnknownTopicOrPartition
+				// If we allow topic creation, now is the point where it is likely that this
+				// is a brand new topic, so try to get metadata on it (which will trigger
+				// the creation process)
+				if b.conf.AllowTopicCreation {
+					_, err := b.fetchMetadata([]string{topic})
+					if err != nil {
+						b.conf.Logger.Info("failed to fetch metadata for new topic %s: %s",
```

Comment:
Logger does not allow formatting. First parameter should be description and than any number of key-value pairs. In this case, it should be 
```go
b.conf.Logger.Info("failed to fetch metadata for new topic", 
    "topic", topic,
     "err", err)
```